src/lib/y2network/dialogs/wireless_wep_keys.rb

```diff
@@ -0,0 +1,253 @@
+          self.value = length_s.empty? ? "128" : length_s
+        end
+
+        def store
+          @builder.key_length = value.to_i
+        end
+
+        def label
+          _("&Key Length")
+        end
+
+        def help
+          _(
+            "<p><b>Key Length</b> defines the bit length of your WEP keys.\n" \
+              "Possible are 64 and 128 bit, sometimes also referred to as 40 and 104 bit.\n" \
+              "Some older hardware might not be able to handle 128 bit keys, so if your\n" \
+              "wireless LAN connection does not establish, you may need to set this\n" \
+              "value to 64.</p>"
+          )
+        end
+      end
+
+      class WEPKeys < CWM::CustomWidget
+        def initialize(settings)
+          textdomain "network"
+          @settings = settings
+        end
+
+        def contents
+          VBox(
+            Table(
+              Id(:wep_keys_table),
+              Header(
+                # Table header label
+                # Abbreviation of Number
+                _("No."),
+                # Table header label
+                _("Key"),
+                # Table header label
+                Center(_("Default"))
+              )
+            ),
+            HBox(
+              # PushButton label
+              PushButton(Id(:wep_keys_add), Yast::Label.AddButton),
+              # PushButton label
+              PushButton(Id(:wep_keys_edit), Yast::Label.EditButton),
+              # PushButton label
+              PushButton(Id(:wep_keys_delete), Yast::Label.DeleteButton),
+              # PushButton label
+              PushButton(Id(:wep_keys_default), _("&Set as Default"))
+            )
+          )
+        end
+
+        # TODO: help text which explain format of WEP keys
+
+        def init
+          refresh_table
+        end
+
+        def refresh_table
+          table_items = @settings.keys.each_with_index.map do |key, i|
+            next unless key
+            Item(Id(i), i.to_s, key, i == @settings.default_key ? "X" : "")
+          end
+
+          Yast::UI.ChangeWidget(Id(:wep_keys_table), :Items, table_items.compact)
+          [:wep_keys_delete, :wep_keys_edit, :wep_keys_default].each do |k|
+            Yast::UI.ChangeWidget(
+              Id(k),
+              :Enabled,
+              !@settings.keys.compact.empty?
+            )
+          end
+          Yast::UI.ChangeWidget(
+            Id(:wep_keys_add),
+            :Enabled,
+            @settings.keys.compact.size < 4 # only 4 keys are possible
+          )
+        end
+
+        # @return [Symbol, nil] dialog result
+        def handle(event)
+          return nil if event["EventReason"] != "Activated"
+
+          cur = Yast::UI.QueryWidget(Id(:wep_keys_table), :CurrentItem).to_i
+          case event["ID"]
+          when :wep_keys_edit
+            key = dialog(@settings.keys[cur])
+            if key
+              @settings.keys[cur] = key
+              refresh_table
```

Comment:
np: Case branches look repetitive. You update the list, refresh the table and set the current element. Probably I would move that logic to a `refresh_table(selected)` method.